source/databricks/settlement_report/settlement_report_job/domain/energy_results/read_and_filter.py

```diff
@@ -47,24 +58,37 @@ def read_and_filter_from_view(
         & (F.col(DataProductColumnNames.time) < args.period_end)
     )
 
-    if args.energy_supplier_ids is not None:
+    if args.calculation_type is CalculationType.BALANCE_FIXING:
+        df = df.where(filter_by_grid_area_codes(args.grid_area_codes))
+        df = read_and_filter_by_latest_calculations(df, args, repository)
+
+    elif args.calculation_id_by_grid_area is not None:
         df = df.where(
-            F.col(DataProductColumnNames.energy_supplier_id).isin(
-                args.energy_supplier_ids
-            )
+            filter_by_calculation_id_by_grid_area(args.calculation_id_by_grid_area)
         )
 
-    if args.calculation_id_by_grid_area is not None:
-        calculation_id_by_grid_area_structs = [
-            F.struct(F.lit(grid_area_code), F.lit(str(calculation_id)))
-            for grid_area_code, calculation_id in args.calculation_id_by_grid_area.items()
-        ]
+    if args.energy_supplier_ids is not None:
```

Comment:
nit: Would this behave the same?
```suggestion
    if args.energy_supplier_ids:
```